query-engine/driver-adapters/executor/src/demo-qc.ts

```diff
@@ -0,0 +1,114 @@
+import * as S from "@effect/schema/Schema";
+import { bindAdapter, ConnectionInfo } from "@prisma/driver-adapter-utils";
+
+import type { DriverAdaptersManager } from "./driver-adapters-manager";
+import { Env } from "./types";
+import * as qc from "./query-compiler";
+import { err } from "./utils";
+import { setupDriverAdaptersManager } from "./setup";
+
+/**
+ * Example run: `DRIVER_ADAPTER="libsql" pnpm demo:se`
+ */
+async function main(): Promise<void> {
+  const env = S.decodeUnknownSync(Env)(process.env);
+  console.log("[env]", env);
+
+  /**
+   * Static input for demo purposes.
+   */
+
+  const url = "file:./db.sqlite";
+
+  const schema = /* prisma */ `
+    generator client {
+      provider = "prisma-client-js"
+    }
+
+    datasource db {
+      provider = "sqlite"
+      url      = "file:./db.sqlite"
+    }
+
+    model User {
+      id Int @id @default(autoincrement())
+      email String @unique
+      name String?
+      posts Post[]
+    }
+
+    model Post {
+      id Int @id @default(autoincrement())
+      title String
+      content String
+      author User @relation(fields: [authorId], references: [id])
+      authorId Int
+    }
+  `;
+
+  const driverAdapterManager = await setupDriverAdaptersManager(env);
+
+  const { compiler: compiler, adapter } = await initQC({
+    env,
+    driverAdapterManager,
+    url,
+    schema,
+  });
+
+  const query = compiler.compile(
+    JSON.stringify({
+      modelName: "User",
+      action: "createOne",
+      query: {
+        arguments: {
+          data: {
+            email: "whatever@gmail.com",
+          },
+        },
+        selection: {
+          id: true,
+        },
+      },
+    }),
+  );
+  console.log("[query]", query);
+}
+
+type InitQueryCompilerParams = {
+  env: Env;
+  driverAdapterManager: DriverAdaptersManager;
+  url: string;
+  schema: string;
+};
+
+async function initQC({
+  env,
+  driverAdapterManager,
+  url,
+  schema,
+}: InitQueryCompilerParams) {
+  const adapter = await driverAdapterManager.connect({ url });
```

Comment:
Random thought: maybe we should put all of the client engine internals (`QueryInterpreter`, `TransactionManager` etc) into a new package, maybe even with some facade around them that `ClientEngine` class in `@prisma/client` would also use if necessary. Then we could easily import that package here to be able to run the engine tests.